src/connection-stage-hooks.tsx

```diff
@@ -144,19 +137,23 @@ export const useConnectionStage = (): {
   const [stage, setStage] = connectionStageContextValue;
   const navigate = useNavigate();
   const connectActions = useConnectActions();
+  const setStatus = useSetConnectStatus();
 
   const actions = useMemo(() => {
     return new ConnectionStageActions(
       connectActions,
       navigate,
       stage,
-      setStage
+      setStage,
+      setStatus
     );
-  }, [connectActions, navigate, stage, setStage]);
+  }, [connectActions, navigate, stage, setStage, setStatus]);
+
+  const status = useConnectStatus(actions.handleConnectionStatus);
 
   const isConnected = useMemo(
```

Comment:
I realise this was in the original but this would be fine (likely faster) without any hook.

Reply:
https://github.com/microbit-foundation/ml-trainer/pull/279/commits/9ecd2f445959c2b0efaf1e840caceee0c336d007